src/uct/base/uct_iface.h

```diff
@@ -396,6 +396,110 @@ typedef struct uct_iface_local_addr_ns {
     }
 
 
+/**
+ * Call TL registration routine.
+ *
+ * @param [in] _name  TL name
+ */
+#define uct_tl_register(_name) \
+    uct_tl_##_name##_ctor()
+
+
+/**
+ * Call TL unregistration routine.
+ *
+ * @param [in] _name  TL name
+ */
+#define uct_tl_unregister(_name) \
+    uct_tl_##_name##_dtor()
+
+
+/**
+ * Register component and TL
+ *
+ * @param [in] _name  Component and TL name
+ * @param [in] ...    Transport and component registration routines
+ *                    attributes (static)
+ */
+#define UCT_COMP_TL_REGISTER(_name, ...) \
+    UCT_COMPONENT_REGISTER_DECL(_name, __VA_ARGS__); \
+    UCT_TL_REGISTER_DECL(_name, __VA_ARGS__); \
+    static void uct_component_tl_##_name##_ctor(void); \
+    void UCS_F_CTOR uct_##_name##_init(void) { \
+        uct_component_tl_##_name##_ctor(); \
+        uct_component_register(_name); \
+        uct_tl_register(_name); \
+    } \
+    void uct_component_tl_##_name##_ctor(void) \
+
+
+/**
+ * Unregister component and TL
+ *
+ * @param [in] _name  Component and TL name
+ * @param [in] ...    Transport and component registration routines
+ *                    attributes (static)
+ */
+#define UCT_COMP_TL_UNREGISTER(_name, ...) \
```

Comment:
maybe just
```
UCT_TL_INIT(_name, __scope) {
  // code
}
```

Reply:
> maybe just
> 
> ```
> UCT_TL_INIT(_name, __scope) {
>   // code
> }
> ```

will not work: user code should be mixed with user initialisation code and ```return``` from user code should not break initialisation sequence

Reply:
I mean to rename the macro to UCT_TL_INIT

Reply:
also, use single arg _scope instead of `__VA_ARGS__`:
1. more readable - it's clear what is the purpose of the argument
2. consistent with kh_init and ucs_array macros
3. can pass empty macro param when static is not needed

Reply:
> I mean to rename the macro to UCT_TL_INIT

renamed to UCT_TL_INIT/CLEANUP

Reply:
> also, use single arg _scope instead of `__VA_ARGS__`:
> 
> 1. more readable - it's clear what is the purpose of the argument
> 2. consistent with kh_init and ucs_array macros
> 3. can pass empty macro param when static is not needed

ok, added _scope parameter